spec.emu

```diff
@@ -15,37 +15,68 @@ contributors: Jordan Harband
 	<emu-alg>
 		1. Let _promise_ be the *this* value.
 		1. If IsPromise(_promise_) is *false*, throw a *TypeError* exception.
-		1. Let _thenFinally_ be ! CreateThenFinally(_onFinally_).
-		1. Let _catchFinally_ be ! CreateCatchFinally(_onFinally_).
+		1. Let _C_ be ? SpeciesConstructor(_promise_, %Promise%).
+		1. Let _thenFinally_ be ! CreateThenFinally(_C_, _onFinally_).
+		1. Let _catchFinally_ be ! CreateCatchFinally(_C_, _onFinally_).
 		1. Return ? Invoke(_promise_, `"then"`, &laquo; _thenFinally_, _catchFinally_ &raquo;).
 	</emu-alg>
 </emu-clause>
 
 <emu-clause id="sec-createthenfinally" aoid="CreateThenFinally">
```

Comment:
I personally would omit the Create*Finally operations since they are only called once. I.e. I would inline them.